features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/base64/Base64LRUCache.kt

```diff
@@ -0,0 +1,42 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.recorder.base64
+
+import android.util.LruCache
+import androidx.annotation.VisibleForTesting
+
+internal object Base64LRUCache : Cache {
```

Comment:
This is nice but let's make it implement also the `ComponentCallbacks2` and listen to the system resources status. Based on the trim level you can decide how eagerly you much memory you could clear from the cache. For more info:
https://developer.android.com/topic/performance/memory
https://github.com/bumptech/glide/blob/fd96de2dd20541c46b3eee24e312969830bc17fb/library/src/main/java/com/bumptech/glide/load/engine/bitmap_recycle/LruBitmapPool.java#L244